amaranth/vendor/xilinx.py

```diff
@@ -484,7 +582,7 @@ def __init__(self, *, toolchain=None):
             else:
                 toolchain = "Vivado"
 
-        assert toolchain in ("Vivado", "ISE", "Symbiflow")
+        assert toolchain in ("Vivado", "ISE", "Symbiflow", "yosys_nextpnr")
```

Comment:
In the Lattice iCE40 platform we use `IceStorm` for toolchain name; in the Lattice ECP5 platform we use `Trellis` for the toolchain name. I think this should probably be `Xray` rather than `yosys_nextpnr`.

Reply:
Done